src/managers/BreakpointManager.ts

```diff
@@ -234,13 +236,21 @@ export class BreakpointManager {
         for (let stagingFilePath in breakpointsByStagingFilePath) {
             promises.push(this.writeBreakpointsToFile(stagingFilePath, breakpointsByStagingFilePath[stagingFilePath]));
         }
+
         await Promise.all(promises);
     }
 
     /**
      * Write breakpoints to the specified file, and update the sourcemaps to match
      */
     private async writeBreakpointsToFile(stagingFilePath: string, breakpoints: BreakpointWorkItem[]) {
+
+        //do not crash if the file doesn't exist
+        if (!fsExtra.existsSync(stagingFilePath)) {
```

Comment:
Can you change this to async `if (!await fsExtra.exists(stagingFilePath)`?Look above around line 237, we call this method for each file all at the same time, so running a synchronous file operation will block other operations from working during that time.

Reply:
I was looking to do that but this method isn't `await` friendly: it still requires a callback.

Reply:
`fsExtra` wraps a bunch of the `fs` methods and makes them promise-ready. Apparently it doesn't do that for `exists`. So check out [fsExtra.pathExists](https://github.com/jprichardson/node-fs-extra/blob/HEAD/docs/pathExists.md) instead. That one has support for promises and does the same thing.

Reply:
Gotcha - fixed.